internal/sim/generators.go

```diff
@@ -43,6 +35,109 @@ func Flip(p float64) Generator[bool] {
 	})
 }
 
+// Numerics
+
+// Nat returns a [Generator] that returns non-negative numbers, also known as
+// natural numbers. Note that Nat does not return all natural numbers
+// equiprobably. Instead, it biases towards numbers closer to zero and other
+// pathological numbers that are more likely to induce bugs (e.g.,
+// math.MaxInt).
+func Nat() Generator[int] {
+	choices := []Weighted[int]{
+		{100, Range(0, 10)},
+		{100, Range(10, 100)},
+		{100, Range(100, 1000)},
+		{100, Range(1000, 10_000)},
+		{100, Range(10_000, 100_000)},
+		{100, Range(100_000, 1_000_000)},
+		{100, Range(1_000_000, 1_000_000_000)},
+		{100, Range(1_000_000_000, 1_000_000_000_000)},
+		{100, Range(1_000_000_000_000, math.MaxInt)},
+		{1, OneOf(
+			math.MaxInt, math.MaxInt-1,
+			math.MaxInt32+1, math.MaxInt32, math.MaxInt32-1,
+			math.MaxInt16+1, math.MaxInt16, math.MaxInt16-1,
+			math.MaxInt8+1, math.MaxInt8, math.MaxInt8-1,
+		)},
+	}
+	return Weight(choices)
+}
+
+// Int returns a [Generator] that returns integers. Note that Int does not
+// return all integers equiprobably. Instead, it biases towards numbers closer
+// to zero and other pathological numbers that are more likely to induce bugs
+// (e.g., math.MaxInt, math.MinInt).
+func Int() Generator[int] {
+	choices := []Weighted[int]{
```

Comment:
This `Weighted` type is brilliant!

Reply:
Yeah, it's very useful! I can't take credit for the idea. I'm mostly just stealing from OCaml's quickcheck library: https://ocaml.org/p/qcheck-core/latest/doc/QCheck/Gen/index.html.